packages/backend-function/src/factory.ts

```diff
@@ -2,137 +2,127 @@ import {
-    const absoluteCodePath = path.isAbsolute(props.codePath)
-      ? props.codePath
-      : path.resolve(getCallerDirectory(new Error().stack), props.codePath);
-    return new AmplifyFunctionFactory({
-      name: props.name,
-      absoluteCodePath,
-      runtime: props.runtime,
-      handler: props.handler,
-    });
+  getInstance = ({
+    constructContainer,
+  }: ConstructFactoryGetInstanceProps): AmplifyFunction => {
+    if (!this.generator) {
+      this.generator = new FunctionGenerator(this.hydrateDefaults());
+    }
+    return constructContainer.getOrCompute(this.generator) as AmplifyFunction;
   };
 
-  /**
-   * Create a function by executing a build command that places build artifacts at a specified location
-   *
-   * TODO: Investigate long-term function building strategy: https://github.com/aws-amplify/amplify-backend/issues/92
-   */
-  static build = async (
-    props: AmplifyFunctionFactoryBuildProps
-  ): Promise<AmplifyFunctionFactory> => {
-    const importPath = getCallerDirectory(new Error().stack);
-
-    await AmplifyFunctionFactory.commandExecutor(props.buildCommand, {
-      cwd: importPath,
-      stdio: 'inherit',
-      shell: 'bash',
-    });
+  private hydrateDefaults = (): HydratedFunctionProps => {
+    return {
+      name: this.resolveName(),
+      entry: this.resolveEntry(),
+    };
+  };
 
-    const absoluteCodePath = path.isAbsolute(props.outDir)
-      ? props.outDir
-      : path.resolve(importPath, props.outDir);
+  private resolveName = () => {
+    // If name is set explicitly, use that
+    if (this.props.name) {
+      return this.props.name;
+    }
+    // If entry is set, use the basename of the entry path
+    if (this.props.entry) {
+      return path.parse(this.props.entry).name;
+    }
 
-    return new AmplifyFunctionFactory({
-      name: props.name,
-      absoluteCodePath,
-      runtime: props.runtime,
-      handler: props.handler,
-    });
+    // Otherwise, use the directory name where the function is defined
+    return path.basename(getCallerDirectory(this.callerStack));
   };
 
-  /**
-   * Creates an instance of AmplifyFunction within the provided Amplify context
-   */
-  getInstance = ({
-    constructContainer,
-  }: ConstructFactoryGetInstanceProps): AmplifyLambdaFunction => {
-    if (!this.generator) {
-      this.generator = new AmplifyFunctionGenerator(this.props);
+  private resolveEntry = () => {
+    // if entry is not set, default to handler.ts
+    if (!this.props.entry) {
+      return path.join(getCallerDirectory(this.callerStack), 'handler.ts');
+    }
+
+    // if entry is absolute use that
+    if (path.isAbsolute(this.props.entry)) {
+      return this.props.entry;
     }
-    return constructContainer.getOrCompute(
-      this.generator
-    ) as AmplifyLambdaFunction;
+
+    // if entry is relative, compute with respect to the caller directory
+    return path.join(getCallerDirectory(this.callerStack), this.props.entry);
```

Comment:
do we have req for these fallbacks?

Reply:
Does req mean requirements? I'm not sure the PMs have explicitly stated it anywhere, but they have written examples where they seem to expect this behavior.